### ISSUE TITLE
[FlexAttention] FLASH backend silently wrong: wrapped gather index misclassified as lane-uniform in vectorized score_mod groups

### ISSUE THREAD

**Issue by drisspg** [pytorch/pytorch#188878]
### 🐛 Describe the bug
ima close and fold into  https://github.com/pytorch/pytorch/pull/188876

Agent log:

The flex_attention FLASH (FA4/CuTeDSL) backend **silently computes wrong results** (~7/8 of elements) for score_mods that combine a relative-position table gather with any other vectorizable captured load — e.g. the natural ALiBi-style pattern:

```python
import torch
from torch.nn.attention.flex_attention import create_block_mask, flex_attention

device, dtype = "cuda", torch.bfloat16
B, H, S, D = 2, 8, 1024, 64
rel_bias = torch.randn(2 * S, device=device)
head_scale = torch.randn(H, device=device)
offset = S - 1

def score_mod(score, b, h, q_idx, kv_idx):
    return score + rel_bias[q_idx - kv_idx + offset] * head_scale[h]

bm = create_block_mask(lambda b, h, q, kv: q >= kv, None, None, S, S,
                       device=device, BLOCK_SIZE=(256, 128))
q, k, v = (torch.randn(B, H, S, D, device=device, dtype=dtype) for _ in range(3))
out = torch.compile(flex_attention)(
    q, k, v, score_mod=score_mod, block_mask=bm, kernel_options={"BACKEND": "FLASH"}
)
# ~92% of elements mismatch vs TRITON backend / eager reference
```

No error is raised; TRITON backend and eager both produce correct results.

### Root cause

The `head_scale[h]` lane-uniform load enables score_mod vectorization (`__vec_size__ = 8`). The `rel_bias` gather index flows through the negative-index wrap in `ModificationWrapperCuteDSL.indirect_indexing`, which returns the wrapped CSE symbol **without registering its semantic index expression** (the non-wrap path does register it). Lane analysis then sees an unknown symbol, which defaults to lane-uniform, and the emitted code loads only lane 0's index and broadcasts it across the group — visible directly in the generated kernel:

```python
tmp6 = cute.where(tmp5 < 0, tmp5 + cutlass.Int32(2048), tmp5)  # per-lane index!
tmp7 = ssa_to_fragment(tmp6, cutlass.Int32)
tmp9 = (in_ptr8[tmp7[0]])              # only lane 0's index is used
for load_idx in cutlass.range(...):
    tmp8[load_idx] = tmp9              # broadcast to all 8 lanes
```

Exactly 7/8 of scores get lane 0's bias → the observed ~92% mismatch. It is also "fast" (issues 1/8 of the loads), so benchmarks without a correctness check look great.

### Fix direction

Register the pre-wrap semantic expression for the wrapped symbol so lane analysis sees the true lane structure, and track wrap-tainted symbols: they keep their per-lane classification but must never classify as *contiguous* (the wrap remaps negative lanes individually, so pre-wrap contiguity does not survive). Per-lane gather loops must also index lane-uniform fragments at `[0]`.

Fix + regression test (fails 91.7% mismatched before, passes after; flex-flash + aux-vectorization suites green, 266 passed on B200) ready on a branch, sibling of the #188871 fixes.

Found while implementing the Sandwich score_mod for attention-gym (meta-pytorch/attention-gym#216); related dtype bugs in the same codegen area: #188871.

### Versions

torch 2.14.0a0+git06b761f (source build 2026-07-02), CUDA 13.3, B200 (SM100), flash-attn CuTeDSL.

cc @ptrblck @msaroufim @eqy @tinglvv @nWEIdia @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo @Chillee @yanboliang @BoyuanFeng @liangel-02 @howardzhang-cv @drisspg

**Comment by drisspg** [comment]
Fuzzed the FLASH captured-buffer load codegen (~100 cases, 4 families: index algebra, multi-dim/indirect tables, buffer layout/dtype, mask_mod + modes; crash-isolated, TRITON oracle, B200) **with this issue's wrap fix applied**. Found two more silent-wrong families that share this issue's root cause — unknown index symbols default to lane-uniform, so vec groups broadcast lane 0's gather:

**1. Untraceable index ops under vec8** — `.abs()`, `.clamp(min=0)`, `torch.minimum` in the index (not in the sympy translation table) become unknown symbols:

```python
tbl = torch.randn(4 * S, device=device); scale = torch.randn(H, device=device)
def mod(s, b, h, q, kv):
    return s + tbl[(q - kv).abs()] * scale[h]   # 72% wrong; correct at vec_size=1
```

**2. Chained gather on the kv axis** — `outer[inner[kv]]`: the contiguous inner load itself promotes vec8 (no explicit promoter needed), and the outer index is a loaded value (`index_expr=None`) → unknown → broadcast:

```python
tbl = torch.randn(8, device=device); doc = torch.randint(0, 8, (S,), device=device)
def mod(s, b, h, q, kv):
    return s + tbl[doc[kv]]   # 59-89% wrong; q-axis chains (tbl2[doc[q], kv]) are fine
```

This is the document-mask-style `table[doc_id[kv]]` pattern, so real-world impact is high. 8 variants reproduce (with/without `* scale[h]`, grad, dynamic shapes, large tables).

**Suggested general fix** rather than patching producers one at a time: default unknown index symbols to GATHER (per-lane); lane-uniform must be *proven* (semantic expr independent of the lane var), never assumed. The wrap-taint fix from this issue becomes a special case of that rule.

Clean under fuzz: affine index algebra (aligned/unaligned offsets, mul/div/mod), computed 2-D/3-D prefixes, non-contiguous/odd/typed buffers, 0-dim scalars, mask_mod captured buffers (packed R2P vec-32 path), GQA/batch/seq-len/dynamic/grad modes.
